crates/pathfinder/src/rpc/types.rs

```diff
@@ -671,9 +671,9 @@ pub mod reply {
     }
 
     /// Describes Starknet's syncing status RPC reply.
-    #[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+    #[derive(Clone, Debug, Serialize, PartialEq)]
     #[serde(untagged)]
-    #[serde(deny_unknown_fields)]
+    #[cfg_attr(test, derive(Deserialize))]
```

Comment:
I didn't see any effect of `deny_unknown_fields` on untagged enum generated code, and I don't think there could be anything, but the deny stopped this from working when applied at the syncing::Status level.

Reply:
Hmm I wonder if I've misunderstood then... Should double-check. My reasoning for this is that the generated deser code only tried each variant and then errored, compared to other versions I remember seeing.

Reply:
Untagged enums have many undocumented peculiarities, for example never use two variants from which one is the prefix of the other. Pity that what you encountered is not documented :( .